Library/Homebrew/tap.rb

```diff
@@ -214,7 +214,8 @@ def install(options = {})
 
     begin
       safe_system "git", *args
-      unless Readall.valid_tap?(self, :aliases => true)
+      if !ARGV.homebrew_developer? &&
+         !Readall.valid_tap?(self, :aliases => true)
```

Comment:
I think we might want to always run `Readall.valid_tap?` because it has side effects (e.g. reporting of invalid formulae) that we don't want to miss. To accommodate the use case of this PR, we could keep the outer condition the same, but make raising the exception conditional:

```ruby
unless Readall.valid_tap?(self, :aliases => true)
  unless ARGV.homebrew_developer?
    raise "Cannot tap #{name}: invalid syntax in tap!"
  end
end
```